fcrepo4/fcrepo4
# Don't allow ldp:membershipResource if you don't have acl:Write

**JIRA Ticket**: https://jira.duraspace.org/browse/FCREPO-1889

# What does this Pull Request do?
Currently you can create a direct or indirect container that would reference a resource you do **not** have `acl:Write` access to with its `ldp:membershipResource` and create triples on that resource.
This now checks for the creation or updating of Direct and Indirect containers to ensure you have `acl:Write` permissions is there is a `ldp:membershipResource` in the request.

# What's new?

@peichman-umd altered the WebACAuthorizingRealm to allow the addition of more than one URI to collect permissions on, this enables us to check both the target of the current request and the referenced URI 

# How should this be tested?

1. Create a container (readonly) and give a user read only access to it.
1. Create a container (users) and give the same user read/write access to it.
1. Try to create an indirect and/or direct container in the _users_ container with a `ldp:membershipResource` of the _readonly_ container. Before PR - works, after PR - 403 forbidden
1. Try to create an indirect and/or direct container in the _users_ container with a `ldp:membershipResource` of another user accessible resource. This should work.
1. Then try to PATCH the indirect/direct container to point the `ldp:membershipResource` to the _readonly_. Before PR - works, after PR - 403 forbidden

There are a lot of integration tests to.

# Additional Notes:

Example:
* Does this change require documentation to be updated? no
* Does this change add any new dependencies? no
* Does this change require any other modifications to be made to the repository (ie. Regeneration activity, etc.)? no
* Could this change impact execution of existing code? no

# Interested parties
@fcrepo4/committers
